lips/lip-20.md

```diff
@@ -0,0 +1,595 @@
+   interchanged must always be well-defined. Any undefined or invalid behaviors
+   such as trying to combine coins of different currency types must never be
+   allowed, and error conditions well specified.
+5. *Conservation, and encapsulation of value*: The value of a coin in
+   circulation can never be accidentally or purposefully altered except through
+   the specified operations defined in this document. Additionally, the
+   `total_value` of all coins minted in a currency must equal the sum of the
+   values of all coins in circulation in that currency no matter the operations
+   performed, with the exception of minting and burning operations.
+6. *Value normalization*: Every currency must have a way to _roughly_ normalize
+   its value to a specified currency so that values expressed across different
+   currencies may be compared for use in the system. In Libra, this specified
+   currency is ≋LBR.
+
+The Libra currency management on-chain is designed to provide a common interface that can
+be used for any currency that ensures that these key properties are kept.
+
+---
+# <a name="registration_and_metadata">Registration of Currencies and Currency Metadata</a>
+---
+
+Every currency on-chain is represented as a Move-defined type `C`. This type can be
+either a resource or struct type. In order for the system to view a type `C` as
+representing a currency on-chain, it must first be _registered_ as such. This may be done by
+calling one of either [`Libra::register_SCS_currency`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-register_scs_currency)
+or [`Libra::register_currency`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-register_currency)
+instantiated with the type `C` that is to be registered as a new SCS or synthetic currency respectively.
+
+## <a name="metadata_spec">Required information for registration of a currency and metadata</a>
+
+When a currency is registered using these functions, a number of metadata need to be
+provided about the currency that is being registered. In particular, every registered
+currency on-chain contains the following data, names in bold are metadata that must be
+provided at the time of registration:
+
+| Name                          | Type                                   | Mutable      | Description                                                                                                                                      |
+| ----------------------------- | -------------------------------------- | ------------ | -----------------------------------------------------------------------------------------------------------------------                          |
+| **to_lbr_exchange_rate**      | FixedPoint32                           | true         | The _rough_ exchange rate from `C` to LBR. Used only for dual attestation threshold and mempool gas price normalization                          |
```

Comment:
Will we still have this at launch?

Reply:
yep! We keep this around so that we won't need to upgrade the `CurrencyInfo` resource, and because this is used for both the dual attestation normalization and gas price normalization.